ligand/views.py

```diff
@@ -44,26 +112,757 @@ class LigandNameSelection(AbsTargetSelection):
+                    q = LigandFingerprint.objects.filter(mfp2__tanimoto=value)
+                    q = q.annotate(sml=TANIMOTO_SML('mfp2', value))
+                    q = q.order_by(TANIMOTO_DIST('mfp2', value))
+                    q = q.values_list('ligand', 'sml')
+                    sql_query = str(q.query).replace('%','%%').replace(string_to_replace_by,'%s')
+                    cursor.execute('SET SESSION rdkit.tanimoto_threshold = %s; ' + sql_query, [similarity_threshold,smiles,smiles,smiles])
+                    cursor_results = cursor.fetchall()
+                    
+
+            elif search_type == 'substructure':
+                if param_dict['stereochemistry']:
+                    stereochemistry = 'true'
+                else:
+                    stereochemistry = 'false'
+                cache_key = "ligand_structural_search_" + ",".join([search_type,smiles,str(stereochemistry),mode])
+                if not(cache_key != False and cache.has_key(cache_key)):
+                    sql_string_placeholder = '^textoreplacebysmiles#'
+                    v = Value(sql_string_placeholder)
+                    if input_type == 'smarts':
+                        m = QMOL(v)
+                    else:
+                        m = MOL(v)
+                    q = LigandMol.objects.filter(molecule__hassubstruct=m).values('ligand')
+                    sql_query = str(q.query).replace(sql_string_placeholder,'%s')
+                    cursor.execute('SET SESSION rdkit.do_chiral_sss=%s; '+sql_query, [stereochemistry,smiles])
+                    cursor_results = cursor.fetchall()
+                    
+            else:
+                raise ValidationError('Unknown ligand structural search parameter: search_type="%s"' % str(search_type))
+            pass
+        # cache.delete(cache_key)
+
+        if mode == 'compact':
+            if not(cache_key != False and cache.has_key(cache_key)):
+                no_results_msg = 'No results found.'
+                if cursor_results != []:
+                    ligand_id = [x[0] for x in cursor_results]
+                    ps = AssayExperiment.objects.filter(ligand__in=ligand_id).prefetch_related('protein', 'ligand', 'ligand__ligand_type','protein__family')
+                    if search_type == 'similarity':
+                        ps = ps.order_by("ligand__id","protein__entry_name")
+                    else:
+                        ps = ps.order_by("ligand__id","protein__entry_name")
+                else:
+                    context = "redirect"
+                    self.request.session['ligand_structural_search_error_msg'] = no_results_msg
+                    self.request.session.modified = True
+                    return context
+                    # return redirect("ligand_selection")                                                                            # 'ligand__ids__web_resource',                                                                      # 'ligand')
```

Comment:
please clean up commented lines

Reply:
Solved in [39bd4a8](https://github.com/protwis/protwis/pull/550/commits/39bd4a8230b189165eddc18d6beb1c8dbe956bd1)